docs/architecture/adr/0019-adopt-web-push.md

```diff
@@ -0,0 +1,111 @@
+come out that are better suited towards synchronization work. The technology or protocol used must
+be usable at large scale in the cloud as well as for self-hosted installations.
+
+Browser extension changes and mandates such as [Manifest V3][mv3] additionally present support
+problems with long-lived background connections.
+
+### Cost and Reliability
+
+New solutions must be able to not only scale to essentially infinite connections but balance cost
+with user growth, all the while delivering high availability. Existing options for mobile
+notifications specifically either have device limits (e.g. [Azure Notification Hubs][hubspricing])
+or lack service level guarantees while not offering modern technologies. Furthermore, some clients
+(e.g. [F-Droid][fdroid]) cannot utilize well-established (albeit proprietary) push backends. A
+single service provider is desired for as much functionality as possible, while still offering
+flexibility for self-host.
+
+## Considered Options
+
+- **Keep the SignalR solution and continue to scale up** - Maintain the cluster of notifications
+  service virtual machines and keep pushing for a larger set to handle scale. Move all mobile
+  notifications to the custom solution and abandon Azure Notification Hubs.
+- **Work around Azure Notification Hubs limits** - Since all mobile devices, even for self-hosted
+  installations, must utilize the Bitwarden cloud for push due to certificate security, devise a
+  solution that shards devices across many Notification Hubs within new subscriptions.
+- **Adopt a new offering for mobile push notifications only** - Use something other than Azure
+  Notification Hubs that doesn't have limits, perhaps with some technology sacrifices.
+- **Adopt a new offering for non-mobile push notifications only** - Use something other than SignalR
+  like a homegrown [Web Push][webpush] backend inside the notifications service.
+- **Adopt a new combined push service provider and utilize Web Push** - Not only migrate away from
+  Azure Notification Hubs but also utilize Web Push with a single, reliable service provider. Host a
+  compatible Web Push backend for self-host with the clients that need it.
+
+## Decision Outcome
+
+Chosen option: **Adopt a new combined push service provider and utilize Web Push**.
+
+### Positive Consequences
+
+- Web Push is [well-supported][caniuse] in most places we need it and is a valuable technology
+  upgrade with ease of maintenance in the future.
+- Several service providers offer Web Push backends for our cloud offering, and the protocol can be
+  implemented within the notifications service for self-host.
+- Web Push fits well into Manifest V3 and service workers.
+- Infrastructure maintenance burdens and cost for the notification service should significantly
+  decrease.
+
+### Negative Consequences
+
+- Need to watch Safari support for Web Push which was just recently released at time of writing.
+- Costs for selecting a service provider.
```

Comment:
Potential cost, but we also have a cost for our current provider. I also believe the provider we found would potentially be less expensive than our current solution.